common/persistence/sql/class.Platform.php

```diff
@@ -148,13 +151,60 @@ public function getNowExpression(){
        // return $this->dbalPlatform->getNowExpression();
        return $date;
     }
+
     /**
-     * 
+     *
      * @author "Lionel Lecaque, <lionel@taotesting.com>"
      * @param string $functionName
+     * @return string
      */
     public function getSqlFunction($functionName){
         return "SELECT " . $functionName . '(?)';
     }
-    
+
+    /**
+     * Returns the SQL snippet to append to any SELECT statement which obtains an exclusive lock on the rows.
+     *
+     * The semantics of this lock mode should equal the SELECT .. FOR UPDATE of the ANSI SQL standard.
+     *
+     * @see https://dev.mysql.com/doc/refman/5.7/en/innodb-locking-reads.html
+     * @see https://www.postgresql.org/docs/9.0/static/sql-select.html#SQL-FOR-UPDATE-SHARE
+     * @return string
+     */
+    public function getWriteLockSQL()

```

Comment:
I learned something today!

Reply:
I'm glad I was able to teach you something new. ;)